aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -392,6 +410,31 @@ impl AptosVM {
         self.move_vm.env.clone()
     }
 
+    /// Returns true if runtime checks for transactions should be performed asynchronously, that
```

Comment:
this one is specifically for a transaction, and internally depends on async_runtime_checks_enabled, which I suppose is set at least per block?

typos: "Tge", "it's" -> its, but in general I hope the comment can be streamlined. Let's add empty line after 1.2. and then say smt like it returns based on a heuristics which determines whether the optimization will have performance benefits for this specific txn. Then we can explain the current heuristic or just say currently it considers whether script or trusted address (since the code itself is very readable and documented).

Reply:
Adjusted